File: Dot examples/ASP Book Examples/Book Examples/Chapter24/CS/App_Code/ShoppingCartHasChanged.cs
using System;
using System.Collections.Generic;
using System.Web.Profile;

namespace AspNetUnleashed
{
    public class ShoppingCartHasChanged
    {
        private bool _hasChanged = false;
        private List<CartItem> _items = new List<CartItem>();

        public bool HasChanged
        {
            get { return _hasChanged; }
            set { _hasChanged = value; }
        }

        public List<CartItem> Items
        {
            get { return _items; }
        }
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter24/VB/App_Code/ShoppingCartHasChanged.vb
Imports System
Imports System.Collections.Generic
Imports System.Web.Profile

Namespace AspNetUnleashed
    Public Class ShoppingCartHasChanged
        Private _hasChanged As Boolean = False
        Private _items As New List(Of CartItem)()

        Public Property HasChanged() As Boolean
            Get
                Return _hasChanged
            End Get
            Set(ByVal Value As Boolean)
                _hasChanged = value
            End Set
        End Property

        Public ReadOnly Property Items() As List(Of CartItem)
            Get
                Return _items
            End Get
        End Property
    End Class
End Namespace
